### ISSUE TITLE
FileDialog in "Open File" mode won't update the title window

### ISSUE THREAD

**Issue by QuanMCPC** [godotengine/godot#62105]
### Godot version

v3.4.4.stable.official [419e713a2]

### System information

Fedora 36 - Linux 5.17.14-300.fc36.x86_64

### Issue description

When creating a new `FileDialog` node and change the mode to `OpenFile` or any other Open operation, the title of the dialog will not changed. If the dialog was changed to `Save` mode then the dialog will rename as intended.

### Steps to reproduce

Step 1. Create a new `FileDialog` node in an existing project or a new project (For demoing, a new project has been created)
![1](https://user-images.githubusercontent.com/60545346/174068232-99615b20-4481-4547-89ce-ab043902cd10.png)

Step 2. Click on the node to open the Inspector
![2](https://user-images.githubusercontent.com/60545346/174068470-ff2a3bb9-ae5c-44be-9e41-7768db0963b6.png)

Step 3. In the `FileDialog` section in the Inspector, change the value of `Mode` to any that said `Open` (Like `OpenFile`, `OpenFiles`, etc.)
![3](https://user-images.githubusercontent.com/60545346/174068599-3f021ec6-97ea-4121-b1ab-2d1e8b73000f.png)

Step 4. In the `WindowDialog` section of the Inspector, change the value of "Window Title" to anything.
![4](https://user-images.githubusercontent.com/60545346/174068735-3486b9b8-6703-4220-8a8b-d65de5921b33.png)

Step 5. To make sure the dialog will open on startup, create a new script in the root node and call `popup()` on the `FileDialog` node in the `_ready()` function (This is for demonstration purposes).
![5](https://user-images.githubusercontent.com/60545346/174068993-c3cd1446-24ac-4fa2-9a1a-7a89e1454436.png)

Step 6. Run the project, and the File Dialog title would still be the `Open...` instead of `Something Some...` like we've set in Step 4.
![6](https://user-images.githubusercontent.com/60545346/174069512-71f91b8a-ddad-4760-9220-36812178652f.png)

### Minimal reproduction project

And here's the project I've used for demonstration.
[aaaaaaaaaa.zip](https://github.com/godotengine/godot/files/8918807/aaaaaaaaaa.zip)

**Comment by Maran23** [comment]
I don't think this is a bug. You just need to disable the first property in `FileDialog`: `Mode Overrides Title`.
![image](https://user-images.githubusercontent.com/66004280/176002669-31efe28f-e9ec-42d5-84ae-592ca38ba4e3.png)

**Comment by Maran23** [comment]
> I think Mode Overrides Title should be modified to only change the title if it wasn't changed from the default previously.

Currently the code is explicit ignoring the title if the `Mode Overrides Title` is on.
I think it makes sense since the property is named like this.
Although we can discuss whether setting a title will automatically override the default mode title, which makes this flag obsolete.

**Comment by gitsomebit** [comment]
I ran into the same issue. I was very confused when I set my FileDialog to OpenFile mode and when I edited the title manually the edit would be reverted on next project start
